### PR TITLE
hmd preview fix

### PR DESCRIPTION
**Pull request by davidkelly**
Several changes here, to allow us to resize interface's window and have the HMD Preview display that properly.  It now letterboxes like the preview on the monitor, updates as you scale the window, etc...  

One caveat:  when you make the window _very_ small, it doesn't look quite right.  This I believe is because I'm using the windowSize which includes the menu area.  However it is used for other size/aspect ratio calculations so for now, it should be ok.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1654/HighFidelity-Beta-PR10961-4b3cf5ee2e97dd12ebdcb2d0357421393b0dc03a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1654/HighFidelity-Beta-PR10961-4b3cf5ee2e97dd12ebdcb2d0357421393b0dc03a.dmg

**Comment by davidkelly** [comment]
This will be tested in the spectator-camera branch, with existing test plan, nothing to qa here...

**Comment by zfox23** [comment]
[Investigate tablet preview bug](https://trello.com/c/aF5PoIs7/251-investigate-tablet-preview-bug)

**Comment by zfox23** [comment]
[Investigate tablet preview bug](https://trello.com/c/aF5PoIs7/251-investigate-tablet-preview-bug)

File: interface/src/ui/ResourceImageItem.cpp [hunk 1]
```diff
@@ -50,32 +54,20 @@ void ResourceImageItemRenderer::onUpdateTimer() {
...
+
+    _url = resourceImageItem->getUrl();
+    _ready = resourceImageItem->getReady();
+    _visible = resourceImageItem->isVisible();
     _window = resourceImageItem->window();
 
-    if (_ready && _visible && !_url.isNull() && !_url.isEmpty() && (visibleChanged || urlChanged || readyChanged || !_networkTexture)) {
-        _networkTexture = DependencyManager::get<TextureCache>()->getTexture(_url);
-    }
+    _networkTexture = DependencyManager::get<TextureCache>()->getTexture(_url);
```
Comment (zfox23) [inline interface/src/ui/ResourceImageItem.cpp]: Ahh...nice and clean here now. :+1:

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 2]
```diff
@@ -514,13 +514,29 @@ void OpenGLDisplayPlugin::renderFromTexture(gpu::Batch& batch, const gpu::Textur
     batch.setPipeline(_presentPipeline);
     batch.draw(gpu::TRIANGLE_STRIP, 4);
     if (copyFbo) {
+        gpu::Vec4i copyFboRect(0, 0, copyFbo->getWidth(), copyFbo->getHeight());
         gpu::Vec4i sourceRect(scissor.x, scissor.y, scissor.x + scissor.z, scissor.y + scissor.w);
-        gpu::Vec4i copyRect(0, 0, copyFbo->getWidth(), copyFbo->getHeight());
+        float aspectRatio = (float)scissor.w / (float) scissor.z; // height/width
```
Comment (zfox23) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: Aspect ratios are normally width/height, no?

Comment (davidkelly) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: I did it this way so I could multiply first.  Agreed that this is really 1/aspectRatio, really.

File: interface/src/ui/ResourceImageItem.cpp (hunk 1 above)
Comment (davidkelly) [inline interface/src/ui/ResourceImageItem.cpp]: all that change checking was useful when update was called in a loop, but now not needed, and somewhat detrimental as it is helpful to always grab the texture when update is called.

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp (hunk 2 above)
Comment (zfox23) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: Got it

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp [hunk 3]
```diff
@@ -242,6 +242,9 @@ void HmdDisplayPlugin::internalPresent() {
 
         glm::ivec4 viewport = getViewportForSourceSize(sourceSize);
         glm::ivec4 scissor = viewport;
+
+        auto fbo = gpu::FramebufferPointer();
```
Comment (zfox23) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Why do you now do this outside the scope of render?

Comment (davidkelly) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: oh -- I was being consistent with viewport and scissor, which are there too.

Comment (zfox23) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Cool. Dev-approved from me!

**Pull request merged**